src/helm/benchmark/scenarios/scenario.py

```diff
@@ -173,26 +174,31 @@ class Scenario(ABC):
     the heavy lifting.
     """
 
-    name: str
+    # Set by the Scenario subclass."""
+    name: str = field(init=False)
     """Short unique identifier of the scenario"""
 
-    description: str
+    # Set by the Scenario subclass."""
+    description: str = field(init=False)
     """Description of the scenario (task, data)"""
 
-    tags: List[str]
+    # Set by the Scenario subclass."""
+    tags: List[str] = field(init=False)
     """Extra metadata (e.g., whether this is a question answering or commonsense task)"""
 
+    # Set by Runner."""
     # TODO: ideally would pass this into `get_instances` to not have to mutate.
-    output_path: str
+    output_path: str = field(init=False, default="")
     """Where downloaded data is cached (to be set by the `Runner`)"""
 
-    def get_definition_path(self) -> str:
-        """Return where the scenario subclass for `self` is defined."""
+    definition_path: str = field(init=False)
+    """Where the scenario subclass for `self` is defined."""
```

Comment:
I wonder since we have readthedocs whether this is really necessary since in the viewer, we can point them to readthedocs (more semantic) than the file in GitHub?  But sometimes looking at the code is nice too since it doesn't lie...

Reply:
We do want this here because we generate readthedocs from attribute docstrings.